File: Algorithm/Python/1000+/1876. Substrings of Size Three with Distinct Characters.py
from collections import defaultdict

class Solution(object):
    def countGoodSubstrings(self, s):
        """
        :type s: str
        :rtype: int
        """
        result = 0
        counter = defaultdict(int)
        for i in range(len(s)):
            counter[s[i]] += 1
            if i > 2:
                c = s[i-3]
                counter[c] -= 1
                if counter[c] == 0:
                    counter.pop(c)
            if len(counter) == 3:
                result += 1
        return result

File: Algorithm/C++/1000+/1876. Substrings of Size Three with Distinct Characters.cc
#include <string>
#include <unordered_map>

class Solution {
public:
    int countGoodSubstrings(std::string s) {
        int result = 0;
        std::unordered_map<char, int> counter;
        for (int i=0; i < s.size(); i++) {
            char c = s[i];
            if (counter.find(c) != counter.end()) {
                counter[c]++;
            } else {
                counter[c] = 1;
            }
            if (i > 2) {
                char c1 = s[i-3];
                counter[c1]--;
                if (counter[c1] == 0) {
                    counter.erase(c1);
                }
            }
            if (counter.size() == 3) {
                result++;
            }
        }
        return result;
    }
};